src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -760,16 +762,15 @@ CHIP_ERROR DiscoveryImplPlatform::StartDiscovery(DiscoveryType type, DiscoveryFi
 
 CHIP_ERROR DiscoveryImplPlatform::StopDiscovery(DiscoveryContext & context)
 {
-    if (!context.GetBrowseIdentifier().HasValue())
+    const auto browseIdentifier = context.GetBrowseIdentifier();
```

Comment:
```suggestion
    const std::optional<intptr_t> browseIdentifier = context.GetBrowseIdentifier();
```
is probably clearer in terms of "this will copy here, so the clear below is ok".